akka-actor-typed/src/main/scala/akka/actor/typed/Logger.scala

```diff
@@ -41,6 +42,63 @@ object LogMarker {
 
 }
 
+/**
+ * Logging options when using `Behaviors.logMessages`.
+ */
+@DoNotInherit
+abstract class LogOptions {
+  /**
+   * User control whether messages are logged or not.  This is useful when you want to have an application configuration
+   * to control when to log messages.
+   */
+  def enabled: Boolean
+
+  /**
+   * The [[akka.event.Logging.LogLevel]] to use when logging messages.
+   */
+  def level: LogLevel
+
+  /**
+   * A [[akka.actor.typed.Logger]] to use when logging messages.
+   */
+  def logger: Logger
+}
+
+/**
+ * Factories for log options
+ */
+object LogOptions {
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  final class LogOptionsImpl private[akka] (val enabled: Boolean, val level: LogLevel, val logger: Logger)
```

Comment:
`private[akka]` in front of `final class`

the other you added is for the constructor and would then not be needed